mne/io/persyst/tests/test_persyst.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+# Authors: Adam Li  <adam2392@gmail.com>
+#          simplified BSD-3 license
+
+import os.path as op
+import shutil
+
+import pytest
+from numpy.testing import assert_array_equal
+
+import mne
+from mne.datasets.testing import data_path, requires_testing_data
+from mne.io import read_raw_persyst
+from mne.io.tests.test_raw import _test_raw_reader
+from mne.utils import run_tests_if_main
+
+fname_lay = op.join(
+    data_path(download=False), 'Persyst',
+    'sub-pt1_ses-02_task-monitor_acq-ecog_run-01_clip2.lay')
+fname_dat = op.join(
+    data_path(download=False), 'Persyst',
+    'sub-pt1_ses-02_task-monitor_acq-ecog_run-01_clip2.dat')
+
+
+@requires_testing_data
+def test_persyst_lay_load():
+    """Test reading Persyst files using path to header file."""
+    raw = read_raw_persyst(fname_lay, preload=False)
+
+    # Test data import
+    assert raw.info['sfreq'] == 200
+    assert raw.preload is False
+
+    # load raw data
+    raw.load_data()
+    assert raw._data.shape == (83, 847)
+    assert raw.preload is True
+
+    # defaults channels to EEG
+    raw = raw.pick_types(eeg=True)
+    assert len(raw.ch_names) == 83
+
+    # no "-Ref" in channel names
+    assert all(['-ref' not in ch.lower()
+                for ch in raw.ch_names])
+
+    # test with preload True
+    raw = read_raw_persyst(fname_lay, preload=True)
+
+
+@requires_testing_data
+def test_persyst_raw():
+    """Test reading Persyst files using path to header file."""
+    raw = read_raw_persyst(fname_lay, preload=False)
+
+    # defaults channels to EEG
+    raw = raw.pick_types(eeg=True)
+
+    # get data
+    data, times = raw.get_data(start=200, return_times=True)
+    assert data.shape == (83, 647)
+
+    # seconds should match up to what is in the file
+    assert times.min() == 1.0
+    assert times.max() == 4.23
+
+    # get data
+    data = raw.get_data(start=200, stop=400)
+    assert data.shape == (83, 200)
+
+    # data should have been set correctly
+    assert not data.min() == 0 and not data.max() == 0
+
+    first_ch_data = raw.get_data(picks=[0], start=200, stop=400)
+    assert_array_equal(first_ch_data.squeeze(), data[0, :])
+
+
+@requires_testing_data
+def test_persyst_wrong_file():
+    """Test reading Persyst files when passed in wrong file path."""
+    with pytest.raises(FileNotFoundError, match='The path you'):
+        read_raw_persyst(fname_dat, preload=True)
+
+    out_dir = mne.utils._TempDir()
```

Comment:
This is the old way to do it. Better to make use of the pytest fixture:
```suggestion
    out_dir = mne.utils._TempDir()
    out_dir = str(tmpdir)
```